.github/workflows/tools/update-issues/issues.js

```diff
@@ -0,0 +1,539 @@
+  if (!matches) {
+    return null;
+  }
+
+  // If there is a tag, append it as a string after a negative one. This will
+  // ensure that versions like "-beta" sort above their production
+  // counterparts.
+  const version = matches[1].split('.').map(x => parseInt(x));
+  if (matches[2]) {
+    version.push(-1);
+    version.push(matches[2]);
+  }
+
+  return version;
+}
+
+/**
+ * Compare two version arrays.  Can be used as a callback to
+ * Array.prototype.sort to sort by version numbers (ascending).
+ *
+ * The last item in a version array may be a string (a tag like "beta"), but
+ * the rest are numbers.  See notes in parseVersion above for details on tags.
+ *
+ * @param {!Array<number|string>} a
+ * @param {!Array<number|string>} b
+ * @return {number}
+ */
+function compareVersions(a, b) {
+  // If a milestone's version can't be parsed, it will be null.  Push those to
+  // the end of any sorted list.
+  if (!a && !b) {
+    return 0;
+  } else if (!a) {
+    return 1;
+  } else if (!b) {
+    return -1;
+  }
+
+  for (let i = 0; i < Math.min(a.length, b.length); ++i) {
+    if (a[i] < b[i]) {
+      return -1;
+    } else if (a[i] > b[i]) {
+      return 1;
+    }
+    // If equal, keep going through the array.
+  }
+
+  // If one has a tag that the other does not, the one with the tag (the longer
+  // one) comes first.
+  if (a.length > b.length) {
+    return -1;
+  } else if (a.length < b.length) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
+
+/**
+ * Compare two Dates.  Can be used as a callback to Array.prototype.sort to
+ * sort by Dates (ascending).
+ *
+ * @param {!Array<number|string>} a
+ * @param {!Array<number|string>} b
+ * @return {number}
+ */
+function compareDates(a, b) {
+  // Sort nulls to the end.
+  if (!a && !b) {
+    return 0;
+  } else if (!a) {
+    return 1;
+  } else if (!b) {
+    return -1;
+  }
+
+  // Date objects can be compared like numbers.
+  if (a < b) {
+    return -1;
+  } else if (a > b) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
+
+/**
+ * Convert a Date to an age in days (by comparing with the current time).
+ *
+ * @param {!Date} d
+ * @return {number} Time passed since d, in days.
+ */
+function dateToAgeInDays(d) {
+  // getTime() and now() both return milliseconds, which we diff and then
+  // convert to days.
+  return (Date.now() - d.getTime()) / (86400 * 1000);
+}
+
+/**
+ * A base class for objects returned by the GitHub API.
+ */
+class GitHubObject {
+  /** @param {!Object} obj */
+  constructor(obj) {
+    /** @type {number} */
+    this.id = obj.id;
+    /** @type {number} */
+    this.number = obj.number;
+
+    if ('created_at' in obj) {
+      /** @type {Date} */
+      this.createdAt = new Date(obj.created_at);
+      /** @type {number} */
+      this.ageInDays = dateToAgeInDays(this.createdAt);
+    }
+
+    if ('closed_at' in obj) {
+      /** @type {Date} */
+      this.closedAt = null;
+      /** @type {number} */
+      this.closedDays = NaN;
```

Comment:
Shouldn't these be the defaults even if the field isn't there?

Reply:
I had thought not.  Some objects don't have a `"closed_at"` field, like comments or events, so I thought it made sense for those not to have those fields at all.  But I can change it to make the subclasses more consistent.